packages/transaction-controller/src/utils.ts

```diff
@@ -207,3 +208,23 @@ export function getAndFormatTransactionsForNonceTracker(
       };
     });
 }
+
+/**
+ * Checks whether a given transaction matches the specified network or chain ID.
+ * This function is used to determine if a transaction is relevant to the current network or chain.
+ *
+ * @param transaction - The transaction metadata to check.
+ * @param chainId - The chain ID of the current network.
+ * @param networkId - The network ID of the current network.
+ * @returns A boolean value indicating whether the transaction matches the current network or chain ID.
+ */
+export function transactionMatchesNetwork(
+  transaction: TransactionMeta,
+  chainId: Hex,
+  networkId: string | null,
+) {
+  if (typeof transaction.chainId !== 'undefined') {
```

Comment:
Could we simplify this logic to something like:

`chainId === transaction.chainId || (!chainId && networkId && networkId === transaction.networkID)`